param.go

```diff
@@ -485,23 +533,85 @@ func newParamGroupedSlice(f reflect.StructField, c containerStore) (paramGrouped
 	return pg, nil
 }
 
-func (pt paramGroupedSlice) Build(c containerStore) (reflect.Value, error) {
-	var itemCount int
+func (pt paramGroupedSlice) getDecoratedValues(c containerStore) []reflect.Value {
+	var items []reflect.Value
+	stores := c.storesToRoot()
+	for _, c := range stores {
+		items = append(items, c.getDecoratedValueGroup(pt.Group, pt.Type.Elem())...)
+	}
+	return items
+}
+
+// searches the given container and its parent for matching group decorators
+// and calls them to commit values. If any of the decorators returns an error,
+// it returns that error immediately. If all decorators succeeds, it returns nil.
+func (pt paramGroupedSlice) callGroupDecorators(c containerStore) error {
+	stores := c.storesToRoot()
+	for _, c := range stores {
+		for _, d := range c.getGroupDecorators(pt.Group, pt.Type.Elem()) {
+			if err := d.Call(c); err != nil {
+				return errParamGroupFailed{
+					CtorID: d.ID(),
+					Key:    key{group: pt.Group, t: pt.Type.Elem()},
+					Reason: err,
+				}
+			}
+		}
+	}
+	return nil
+}
+
+// searches the given container and its parent for matching group providers and
```

Comment:
suggestion: Search the given container and its parent for matching group providers and call them to commit values.
Return the number of providers called and an a non-nil error from the first provided, if an error is encountered.